src/docs/stan-reference/functions.tex

```diff
@@ -3424,6 +3424,127 @@ \chapter{Mixed Operations}\label{mixed-operations.chapter}
+of the algebraic system.
+
+\section{Call to the Algebraic Solver}
+
+\begin{description}
+%
+   \fitemthreelines{real[~]}{algebra\_solver}%
+                  {function \farg{algebra\_system}}%
+                  {real[] \farg{initial\_guess}}%
+                  {real \farg{theta}, real[] \farg{x\_r}, int[] \farg{x\_i}}%
+                  {Solves the algebraic system, given an initial guess,
+                    using the Powell Hybrid algorithm with the unsupported
+                    implementation from Eigen.}
+                   
+\fitemfourlines{real[~]}{algebra\_solver}%
+                  {function \farg{algebra\_system}}%
+                  {real[] \farg{initial\_guess}}%
+                  {real \farg{theta}, real[] \farg{x\_r}, int[] \farg{x\_i}}%
+                  {real \farg{rel\_tol}, real \farg{f\_tol}, int \farg{max\_steps}}%
+                  {Solves the algebraic system, given an initial guess,
+                    using the Powell Hybrid algorithm with the unsupported
+                    implementation from Eigen with additional control
+                    parameters for the solver.}
+%
+\end{description}
+
+
+\subsection{Arguments to the Algebraic Solver}
+
+The arguments to the algebraic solver are as follows:
+%
+\begin{enumerate}
+\item \farg{algebra\_system}: function literal refering to a function
+  specifying the system of algebraic equations with signature described
+  in \refsection{functions-algebra-function}:
+  \begin{quote}
+    (real[], real[], real[], int[]):real[]
+  \end{quote}
+  The arguments represent (1) unknown, (2) parameters, (3) real data, and
+  (4) integer data, and the return value contains the value of the algebraic
+  function, which goes to 0 when we plug in the solution to the algebraic system,
+\item \farg{initial\_guess}: initia guess for the solution, type \code{real[]},
+\item \farg{theta}: parameters, type \code{real[]},
+\item \farg{x\_r}: real data, type \code{real[]}, data only, and
+\item \farg{x\_i}: parameters, type \code{int[]}, data only.
+\end{enumerate}
+
+For more fine-grained control of the algebraic solver, these parameters can also
+be provided:
+%
+\begin{enumerate}
+  \addtocounter{enumi}{5}
+  \item \farg{rel\_tol}: relative tolerance for the algebraic solver, type \code{real}, data only,
+  \item \farg{function\_tol}: function tolerance for the algebraic solver, type \code{real}, data only,
+  \item \farg{max\_num\_steps}: maximum number of steps to take in the algebraic solver, type \code{int}, data only.
+\end{enumerate}
+
+\subsection{Return Value}
+The return value for the algebraic solver is an array of type \code{real[]}, with values
+which, when plugged in as the \code{unknowns} make the algebraic function go to 0.
+
+\subsection{Sizes and Parallel Arrays}
+Certain sizes must match. In particular, the following groups are of the same size:
+%
+\begin{enumerate}
+  \item unknowns passed into the system function, algebraic function returned by the system
```

Comment:
This is confused.  The second group isn't things that have to be the same sizes.  I'd skip the enumeration and say

```
Certain sizes have to be consistent. The initial guess, return value of the solver, and return value of the algebraic function must all be the same size. 

The parameters, real data, and integer data will be passed from the solver directly to the system function. 
```